### PR TITLE
drivers: wifi: nxp:Fix the compilation error on the MIMXRT1170_EVK board with the NXP_88W8987_MURATA_1ZM_M2 WiFi module

### LINKED ISSUE #81478: Compilation Error with mimxrt1170_evk Board and NXP_88W8987_MURATA_1ZM_M2 WiFi Module
**Describe the bug**
When compiling the Zephyr sample project for the mimxrt1170_evk board with the NXP_88W8987_MURATA_1ZM_M2 WiFi module, the build process fails due to a missing `board.h` file.

**To Reproduce**

Run the following command
```
git apply patch.txt
west build -b mimxrt1170_evk/mimxrt1176/cm7 zephyr/samples/net/wifi/shell
```

**Expected behavior**
The build process should complete successfully without any errors, and the resulting binary should be able to communicate with the WiFi module.

**Impact**
This issue is a showstopper as it prevents the successful compilation of the Zephyr sample project for the mimxrt1170_evk board with the specified WiFi module.

**Logs and console output**
```
/home/csbobo/code/zephyr_code/modules/hal/nxp/mcux/middleware/wifi_nxp/sdio_nxp_abs/mlan_sdio.c:11:10: fatal error: board.h: No such file or directory
11 | #include <board.h>
|          ^~~~~~~~~
```
**Environment (please complete the following information):**
 - OS: [Linux]
 - Toolchain: [Zephyr SDK]
 - Commit SHA or Version used: [8469084dfae85f854555f0607f2c838dad097235]

**Additional context**
The mimxrt1170_evk board is being used with the NXP_88W8987_MURATA_1ZM_M2 WiFi module. The issue occurs regardless of the WiFi module's configuration or the board's setup.

**Attachments**
[bug.txt](https://github.com/user-attachments/files/17784168/bug.txt)

[patch.txt](https://github.com/user-attachments/files/17784169/patch.txt)

```[tasklist]
### Tasks
```

### PR DESCRIPTION
**Pull request by CsBoBoNice**
- Fix the compilation error on the MIMXRT1170_EVK board with the NXP_88W8987_MURATA_1ZM_M2 WiFi module
- samples net wifi shell add mimxrt1170_evk support

### CODE REVIEW

**Comment by CsBoBoNice** [comment]
This code needs to be used with the following pull request:

https://github.com/zephyrproject-rtos/hal_nxp/pull/473

File: samples/net/wifi/shell/boards/mimxrt1170_evk_mimxrt1176_cm7_A.overlay [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+&usdhc1 {
+    #address-cells = <1>;
+    #size-cells = <0>;
+    status = "okay";
+    pinctrl-0 = <&pinmux_usdhc1>;
+    pinctrl-1 = <&pinmux_usdhc1_dat3_nopull>;
+    pinctrl-names = "default", "nopull";
+    pwr-gpios = <&gpio10 2 GPIO_ACTIVE_LOW>;
+    max-current-180 = <1000>;  /* 设置1.8V下最大电流为1000mA */
```
Comment (mmahadevan108) [inline samples/net/wifi/shell/boards/mimxrt1170_evk_mimxrt1176_cm7_A.overlay]: Please fix language in comments.

**Comment by danieldegrasse** [comment]
> This code needs to be used with the following pull request:
> 
> [zephyrproject-rtos/hal_nxp#473](https://github.com/zephyrproject-rtos/hal_nxp/pull/473)

There is a process documented for linking module updates with Zephyr changes. Please take a look at the documentation here: https://docs.zephyrproject.org/latest/develop/modules.html#process-for-submitting-changes-to-existing-modules and modify the `west.yml` to point to the HAL PR like so:

```diff
diff --git a/west.yml b/west.yml
index 87287fb39ac..84df6002a95 100644
--- a/west.yml
+++ b/west.yml
@@ -198,7 +198,7 @@ manifest:
       groups:
         - hal
     - name: hal_nxp
-      revision: 3c64cd63125c86870802a561ce79dc33697b005c
+      revision: pull/473/head
       path: modules/hal/nxp
       groups:
         - hal
```

File: drivers/wifi/nxp/Kconfig.nxp [hunk 2]
```diff
@@ -314,43 +314,69 @@ config NXP_WIFI_SHELL
 
 menu "Wi-Fi driver Stack configurations"
 
-config NXP_WIFI_MON_TASK_STACK_SIZE
+config NXP_WIFI_MON_STACK_SIZE
```
Comment (danieldegrasse) [inline drivers/wifi/nxp/Kconfig.nxp:317]: Why rename these Kconfigs/change defaults? Ping @MaochenWang1 to take a look at this.

File: samples/net/wifi/shell/boards/mimxrt1170_evk_mimxrt1176_cm7_A.conf [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=4096
+CONFIG_NET_MGMT_EVENT_STACK_SIZE=4096
+CONFIG_NET_SOCKETS_SERVICE_STACK_SIZE=4096
+
+# WiFi Core Configuration
+CONFIG_WIFI=y
+CONFIG_WIFI_LOG_LEVEL_ERR=y
+CONFIG_WIFI_NXP=y
+CONFIG_NXP_88W8987=y
+CONFIG_NXP_88W8987_MURATA_1ZM_M2=y
```
Comment (danieldegrasse) [inline samples/net/wifi/shell/boards/mimxrt1170_evk_mimxrt1176_cm7_A.conf:36]: I don't *love* the current approach we use for module selection here (IE choosing it with a Kconfig). This is a property of the hardware- IMO ideally we should be using a different devicetree compatible for each module.

The other issue here is that this M.2 slot can support a variety of modules, not just the 1ZM. There is also an IW416 based module for example (1XK).

File: drivers/wifi/nxp/Kconfig.nxp
Comment (mmahadevan108) [inline drivers/wifi/nxp/Kconfig.nxp:1]: Please fix the commit messages: https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-guidelines

**Review by danieldegrasse — COMMENTED** [review]
Thanks for this contribution. A few general comments:
- commit messages and comments need to be cleaned up to be entirely in English
- please link the update to the NXP HAL module as mentioned in my other comment
- We should add some documentation to the RT1170 board page to indicate support for this WiFi module is validated with Zephyr
- Please take a look at the failing compliance checks and annotations about formatting issues, these will need to be resolved (along with the failing CI tests)

Also, just wanted to note that with a few HAL changes (adding the IW416 firmware blobs and changing the CMakeLists to include them), as well as changing the card detection method, I was able to get the AM510-USD module working.

**Comment by dleach02** [comment]
ping @MaochenWang1

File: drivers/wifi/nxp/Kconfig.nxp (hunk 2 above)
Comment (dleach02) [inline drivers/wifi/nxp/Kconfig.nxp:317]: Renaming Kconfigs are not something that can be done as they are already established in the project. 

Please revert this aspect of your PR to the previous names.

**Comment by CsBoBoNice** [comment]
Thank you very much for your feedback and guidance! I am deeply sorry for my negligence and poorly considered code. I will make the necessary modifications according to the issues you have pointed out.

File: drivers/wifi/nxp/Kconfig.nxp (hunk 2 above)
Comment (MaochenWang1) [reply drivers/wifi/nxp/Kconfig.nxp:317]: Don't need to rename these Kconfigs. I have a Zephyr side PR to change the thread stack size and priority:
https://github.com/zephyrproject-rtos/zephyr/pull/81693
The corresponding hal_nxp(wifi_nxp)  change was done for downstream, but not upstream yet.

File: drivers/wifi/nxp/nxp_wifi_drv.c [hunk 4]
```diff
@@ -1197,18 +1197,23 @@ static int nxp_wifi_power_save(const struct device *dev, struct wifi_ps_params *
 			}
 			break;
 		case WIFI_PS_PARAM_MODE:
+			
 			if (params->mode == WIFI_PS_MODE_WMM) {
+				#if CONFIG_WMM_UAPSD
```
Comment (MaochenWang1) [inline drivers/wifi/nxp/nxp_wifi_drv.c:1202]: Please use:
#ifdef CONFIG_NXP_WIFI_WMM_UAPSD

**Review by MaochenWang1 — CHANGES_REQUESTED** [review]

File: drivers/wifi/nxp/nxp_wifi_drv.c (hunk 4 above)
Comment (CsBoBoNice) [reply drivers/wifi/nxp/nxp_wifi_drv.c:1202]: Sure, I will make the changes according to your request.
There is a modification that I need your help with:
Regarding `hal_nxp/blob/master/zephyr/module.yml` , I don't know how to add `sduart8987_wlan_bt.bin.inc` or other binary files needed in `hal/nxp/mcux/middleware/wifi_nxp/wifi_bt_firmware`. Could you please help add them to `module.yml` ?
https://github.com/zephyrproject-rtos/hal_nxp/blob/master/zephyr/module.yml#L19

Comment (CsBoBoNice) [reply drivers/wifi/nxp/nxp_wifi_drv.c:1202]: There is another question I would like to ask, also regarding the `modules/hal/nxp/zephyr/module.yml` file:
From the code in `samples/net/wifi/shell/sample.yaml`, it seems that `samples/net/wifi/shell` currently supports `frdm_rw612` and `rd_rw612_bga`.
I used the following commands to compile:
```
west update
```

```
west build -b frdm_rw612 zephyr/samples/net/wifi/shell/
```
or
```
west build -b rd_rw612_bga zephyr/samples/net/wifi/shell/
```

Both commands result in the following error:
```
CMake Error at /home/csbobo/code/zephyr_code/modules/hal/nxp/zephyr/src/rw61x/CMakeLists.txt:45 (message):
  Couldn't find signed firmware!
  /home/csbobo/code/zephyr_code/modules/hal/nxp/zephyr/blobs/rw61x/rw61x_sb_wifi_a2.bin
```

In `modules/hal/nxp/zephyr/module.yml`, there is already a description related to `rw61x_sb_wifi_a2.bin`. What command can be used to obtain `rw61x_sb_wifi_a2.bin` and allow `frdm_rw612` and `rd_rw612_bga` to compile normally?

Comment (MaochenWang1) [reply drivers/wifi/nxp/nxp_wifi_drv.c:1202]: Please use cmd:
`west blobs fetch hal_nxp`

**Comment by CsBoBoNice** [comment]
Dear contributors,

I would like to inquire if anyone would be able to assist in moving this PR forward? This PR is currently critical to the development of my project. Should there be any aspects of my code implementation that do not conform to the project's standards or plans, feel free to make the necessary corrections on another PR. However, please do not overlook this crucial bug fix.
Additionally, I would like to report a related issue: on the MIMXRT1170_EVK development board, the NXP_88W8987_MURATA_1ZM_M2 module can be driven normally through the usdhc1 interface. However, when I attempt to use the usdhc2 interface, I encounter issues that prevent it from functioning correctly. Due to the unresolved bug in the current PR, I am unable to submit a detailed report on the usdhc2 driver issue. I kindly ask for everyone's help so that I can continue with the development of the project.

Thank you very much for your attention and support. I look forward to your replies.

**Comment by jukkar** [comment]
People at NXP, any progress on this?

**Comment by MaochenWang1** [comment]
FYI @sandip-dalvi

**Comment by MaochenWang1** [comment]
Hi @CsBoBoNice  please resolve the conflict on zephyr and hal_nxp side

**Pull request closed without merging**